dartagnan/src/main/java/com/dat3m/dartagnan/expression/BNonDet.java

```diff
@@ -42,10 +43,8 @@ public BooleanFormula toBoolFormula(Event e, SolverContext ctx) {
 	@Override
 	public boolean getBoolValue(Event e, Model model, SolverContext ctx) {
 		Boolean value = model.evaluate(toBoolFormula(e, ctx));
-		if(value != null) {
-			return value;
-		}
-		throw new RuntimeException("No value in the model for " + this);
+		Preconditions.checkNotNull(value, "No value in the model for " + this);
```

Comment:
`InvalidArgumentException` is better here (although not ideal). So use `Preconditions.checkArgument(value != null, ...)` instead.